### PR TITLE
add easyblock for pplacer

### PR DESCRIPTION
**Pull request by boegel**
cfr. https://github.com/hpcugent/easybuild-easyconfigs/pull/1056

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86: please review?

File: easybuild/easyblocks/p/pplacer.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+##
+# Copyright 2016-2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: FWO

File: easybuild/easyblocks/p/pplacer.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from easybuild.tools.run import run_cmd
+
+
+sanity_check_paths = {
+    'files': ['bin/guppy', 'bin/pplacer', 'bin/rppr'],
+    'dirs': [],
+}
+
+sanity_check_commands = [('pplacer', '--version | grep %(version)s')]
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: first imports, then global vars?

File: easybuild/easyblocks/p/pplacer.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+# see http://matsen.github.io/pplacer/compiling.html#opam
+class EB_pplacer(ConfigureMake):
+    """Support for building/installing pplacer."""
+
+    def configure_step(self):
+        """Custom configuration procedure for pplacer."""
+        # install dir has to be non-existing when we start (it may be there from a previous (failed) install
+        if os.path.exists(self.installdir):
+            shutil.rmtree(self.installdir)
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: debug warning about this

File: easybuild/easyblocks/p/pplacer.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+        super(EB_pplacer, self).build_step()
+
+    def install_step(self):
+        """Custom install procedure for pplacer: copy 'bin' directory to install directory."""
+        from_bindir = os.path.join(self.builddir, 'pplacer-%s' % self.version, 'bin')
+        to_bindir = os.path.join(self.installdir, 'bin')
+        try:
+            shutil.copytree(from_bindir, to_bindir)
+
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: drop empty line?

File: easybuild/easyblocks/p/pplacer.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        env.setvar('OCAML_BACKEND', 'gcc')
+
+        run_cmd("eval `opam config env` && cat opam-requirements.txt | xargs opam install -y")
+
+        txt = "let version = \"v%s\"\n" % self.version
+        write_file(os.path.join(self.builddir, 'pplacer-%s' % self.version, 'common_src', 'version.ml'), txt)
+
+    def build_step(self):
+        """Custom build procedure for pplacer: set up OPAM environment and run 'make'."""
+
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: less white space in the function?

File: easybuild/easyblocks/p/pplacer.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+import os
+import shutil
+
+import easybuild.tools.environment as env
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import write_file
+from easybuild.tools.run import run_cmd
+
+
+sanity_check_paths = {
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: why is this here?

File: easybuild/easyblocks/p/pplacer.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/p/pplacer.py]: copy-paste screwup, so that's where it went :P

File: easybuild/easyblocks/p/pplacer.py (hunk 6 above)
Comment (boegel) [inline easybuild/easyblocks/p/pplacer.py]: it shouldn't be, copy-paste gone bad

File: easybuild/easyblocks/p/pplacer.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        # configure OPAM to install pplacer dependencies
+        env.setvar('OPAMROOT', self.installdir)
+        run_cmd("opam init")
+
+        run_cmd("opam repo add pplacer-deps %s/pplacer-opam-repository*/" % self.builddir)
+        run_cmd("opam update pplacer-deps")
+
+        env.setvar('OCAML_BACKEND', 'gcc')
+
+        run_cmd("eval `opam config env` && cat opam-requirements.txt | xargs opam install -y")
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: no harm in adding the `-t` option to xargs here.

**Comment by boegel** [comment]
@wpoely86: remarks fixed

File: easybuild/easyblocks/p/pplacer.py [hunk 8]
```diff
@@ -0,0 +1,92 @@
+##
+# Copyright 2016-2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# # Flemish Research Foundation (FWO) (http://www.fwo.be/en)
```
Comment (wpoely86) [inline easybuild/easyblocks/p/pplacer.py]: drop one #

**Comment by wpoely86** [comment]
:+1:

**Comment by boegel** [comment]
Tested with https://github.com/hpcugent/easybuild-easyconfigs/pull/1056, works as intended.

Thanks for the feedback @wpoely86!
